### ISSUE TITLE
Network policy is too restrictive

### ISSUE THREAD

**Issue by briantopping** [gardener/gardener#3329]
Hello experts, I'm new to Gardener so I don't know if this is an issue or not, reporting on what I did to get my first cluster working. During shoot reconciliation, I was having a problem with gardenlet reaching the shoot apiserver. Removing this aspect of the NetworkPolicy allowed it to continue reconciling

https://github.com/gardener/gardener/blob/2f725a489a03254397e16c4950d1c7e544a1c610/charts/seed-controlplane/charts/kube-apiserver/templates/networkpolicy-allow-kube-apiserver-policy.yaml#L41

Not entirely sure if there are more requirements that I missed on the security of this endpoint. Thoughts?

**Comment by timebertt** [comment]
Hmm, that sounds odd. This particular line is supposed to allow traffic from all other control plane components to their API server. It shouldn't restrict anything and also not the gardenlet to talk to the API server...

**Comment by briantopping** [comment]
What if it's allowing other control plane components but not other namespaces?

I wonder if this is a behavioral difference between how Calico and Cilium interpret the netpol?

**Comment by briantopping** [comment]
What appears to be happening is Cilium distills criteria for netpol comparisons to labels and what I guess I’d call "set semantics". One of the problems I was seeing was the connection to the seed api-server was being labelled as `reserved:host`, which is basically any address on the actual machine. This didn’t match with any of the labels that were defined by the netpol so none of the rules matched. I see how to change things with `CiliumNetworkPolicy`, but this is obviously not portable. For now, I figured out a way to put Cilium in a kind of “permissive” mode like one would do with SELinux.

Closing in deference to a more specific identification of a the problem (and hopefully a strategy or resolution) at a later date.
